### PR TITLE
fix: reproduce v4 behavior for data compatibility

### PR DESCRIPTION
**Pull request by alexandrebodin**
### What does it do?

Fix https://github.com/strapi/strapi/issues/20948

- This only ensure the v4 behavior still exists to avoid any data loss. The app is already logging a warning for the user to change their config. 
- It also fixes the ctb behavior for new relations

- This will need another refactoring at a later time

### Related issue(s)/PR(s)

Implements DX-1512

#20948

### CODE REVIEW

**Comment by innerdvations** [comment]
Failing test just because it's trying to match metadata and doesn't recognize the new attribute

**Review by Marc-Roig — APPROVED** [review]
approach looks good to me 👍

**Review by innerdvations — COMMENTED** [review]
Makes sense, and it worked for me.

I'm not sure what the best way would be, but it would be nice to have an explicit test for this. Maybe an API test with a "bad" schema, and check that the expected tables exist?

**Comment by alexandrebodin** [comment]
@innerdvations I added a test :)

**Review by innerdvations — APPROVED** [review]

File: tests/api/core/database/v4-self-ref-compat.test.api.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      type: 'relation',
+      relation: 'manyToMany',
+      target: 'api::test.test',
+      inversedBy: 'parents',
+    },
+    parents: {
+      type: 'relation',
+      relation: 'manyToMany',
+      target: 'api::test.test',
+      inversedBy: 'children', // intentionnaly wrong to validate retro compatibility
```
Comment (innerdvations) [inline tests/api/core/database/v4-self-ref-compat.test.api.js:26]:
```suggestion
      inversedBy: 'children', // intentionally wrong to validate retro compatibility
```

**Pull request merged**